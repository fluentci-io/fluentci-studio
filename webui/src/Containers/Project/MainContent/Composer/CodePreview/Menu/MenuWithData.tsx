import Menu from "./Menu";
import { FC } from "react";
import copyToClipboard from "copy-to-clipboard";
import { useRecoilValue } from "recoil";
import { CodePreviewState } from "../CodePreviewState";
import { PlateformSelectState } from "../PlateformSelect/PlateformSelectState";

const MenuWithData: FC = () => {
  const [{ filename }] = useRecoilValue(PlateformSelectState);
  const code = useRecoilValue(CodePreviewState);
  const onCopy = () => {
    copyToClipboard(code);
  };

  const onDownload = () => {
    const blob = new Blob([code], { type: "text/yaml" });
    const url = URL.createObjectURL(blob);
    const a = document.createElement("a");
    a.href = url;
    a.download = filename;
    document.body.appendChild(a);
    a.click();
    document.body.removeChild(a);
    URL.revokeObjectURL(url);
  };

  return <Menu onCopy={onCopy} onDownload={onDownload} />;
};

export default MenuWithData;
