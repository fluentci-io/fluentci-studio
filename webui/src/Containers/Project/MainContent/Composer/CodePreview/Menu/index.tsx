import Menu from "./MenuWithData";

export default Menu;
