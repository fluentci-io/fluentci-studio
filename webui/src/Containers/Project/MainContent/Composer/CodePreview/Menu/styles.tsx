import styled from "@emotion/styled";

export const Container = styled.div`
  position: absolute;
  right: 20px;
`;

export const Button = styled.button`
  background-color: initial;
  border: none;
  cursor: pointer;
  color: #ffffffa3;
  &:hover {
    color: #fff;
  }
`;
