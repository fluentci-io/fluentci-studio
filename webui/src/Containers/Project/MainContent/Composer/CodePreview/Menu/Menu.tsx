import { FC } from "react";
import { Download } from "@styled-icons/bootstrap";
import { Copy } from "@styled-icons/ionicons-outline";
import { Button, Container } from "./styles";

export type MenuProps = {
  onDownload: () => void;
  onCopy: () => void;
};

const Menu: FC<MenuProps> = ({ onCopy, onDownload }) => {
  return (
    <Container>
      <Button onClick={onCopy}>
        <Copy size={18} />
      </Button>
      <Button onClick={onDownload}>
        <Download size={18} />
      </Button>
    </Container>
  );
};

export default Menu;
