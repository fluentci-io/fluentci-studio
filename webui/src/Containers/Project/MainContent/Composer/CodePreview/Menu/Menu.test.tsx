import { render } from "@testing-library/react";
import Menu from "./Menu";

describe("Menu", () => {
  it("should render the component", () => {
    const { container } = render(
      <Menu onCopy={vi.fn()} onDownload={vi.fn()} />
    );
    expect(container).toMatchSnapshot();
  });
});
