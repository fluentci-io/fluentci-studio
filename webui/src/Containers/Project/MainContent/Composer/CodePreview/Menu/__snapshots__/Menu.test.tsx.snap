// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Menu > should render the component 1`] = `
<div>
  <div
    class="css-868fxl"
  >
    <button
      class="css-19nl8qs"
    >
      <svg
        aria-hidden="true"
        class="StyledIconBase-sc-ea9ulj-0 jvZCGA"
        fill="currentColor"
        focusable="false"
        height="18"
        viewBox="0 0 512 512"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="336"
          rx="57"
          ry="57"
          stroke="currentColor"
          stroke-linejoin="round"
          stroke-width="32"
          width="336"
          x="128"
          y="128"
        />
        <path
          d="m383.5 128 .5-24a56.16 56.16 0 0 0-56-56H112a64.19 64.19 0 0 0-64 64v216a56.16 56.16 0 0 0 56 56h24"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="32"
        />
      </svg>
    </button>
    <button
      class="css-19nl8qs"
    >
      <svg
        aria-hidden="true"
        class="StyledIconBase-sc-ea9ulj-0 jvZCGA"
        fill="currentColor"
        focusable="false"
        height="18"
        viewBox="0 0 16 16"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"
        />
        <path
          d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"
        />
      </svg>
    </button>
  </div>
</div>
`;
